File: app/src/main/java/nejati/me/omdbapi/view/adapter/StatePagerAdapter.java
package nejati.me.omdbapi.view.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

/**
 * Authors:
 * Reza Nejati <reza.n.j.t.i@gmail.com>
 * Copyright © 2017
 */
public class StatePagerAdapter  extends FragmentStatePagerAdapter {
    private int mCurrentPosition = -1;

    private final ArrayList<Fragment> mFragmentList = new ArrayList<>();
    private final ArrayList<String> mFragmentTitleList = new ArrayList<>();

    public StatePagerAdapter(@NonNull FragmentManager fm, int behavior) {
        super(fm, behavior);
    }

    @NotNull
    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }
    @Override
    public CharSequence getPageTitle(int position)
    {
        return mFragmentTitleList.get(position);
    }


    @Override
    public int getCount() {
        return mFragmentList.size();
    }
    public void addFrag(String title,Fragment fragment)
    {
        mFragmentList.add(fragment);
        mFragmentTitleList.add(title);
    }
}

File: app/src/main/java/nejati/me/omdbapi/view/adapter/StatePagerAdapter.kt
package nejati.me.omdbapi.view.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import nejati.me.omdbapi.view.FragmentModel
import java.util.*

/**
 * Authors:
 * Reza Nejati <reza.n.j.t.i></reza.n.j.t.i>@gmail.com>
 * Copyright © 2017
 */
class StatePagerAdapter(
    fm: FragmentManager,
    behavior: Int,
    private val mFragmentList: MutableList<FragmentModel>
) : FragmentStatePagerAdapter(fm, behavior) {
    override fun getItem(position: Int): Fragment {
        return mFragmentList[position].fragment
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return mFragmentList[position].title
    }

    override fun getCount(): Int {
        return mFragmentList.size
    }

}